lib/src/pick_datetime.dart

```diff
@@ -235,3 +247,57 @@ const _months = {
   'Nov': 11,
   'Dec': 12,
 };
+
+/// This returns a 4-digit year from 2-digit input
+///
+/// For years 0-49 it returns 2000-2049
+/// For years 50-99 it returns 1950-1999
+///
+/// Logic taken from:
+/// https://www.ietf.org/rfc/rfc2822.txt
+int _normalizeYear(int year) {
+  if (year < 100) {
+    if (year < 50) {
+      return 2000 + year;
+    } else {
+      return 1900 + year;
+    }
+  }
+  return year;
+}
+
+/// The Duration to add to a DateTime to get the correct time in UTC
+Duration _getTimeZoneOffset(String? timeZone) {
+  if (timeZone == null) return Duration.zero;
+  if (RegExp(r'^[+-]\d{4}$').hasMatch(timeZone)) {
+    final operator = timeZone[0] == '-' ? 1 : -1;
+    final hours = timeZone.substring(1, 3);
+    final minutes = timeZone.substring(3, 5);
+    return Duration(
+      hours: int.parse(hours) * operator,
+      minutes: int.parse(minutes) * operator,
+    );
+  } else {
+    final timeZoneOffset = _timeZoneOffsets[timeZone] ?? 0;
```

Comment:
Due to the fallback `?? 0` we treat every unknown time zone abbreviation as `UTC`. I expect an exception similar to what `DateTime.parse()` would throw

```
FormatException: Invalid date format
2023-01-09T12:31:54XXX
```

Reply:
That means the package refuses to parse when not recognizing the time zone string. Is this really desired?
Also, the method only gets `String? timeZone`, not the whole String. Should there be a second, named, parameter for the whole String, so the exception can show it?